### PR TITLE
feat(calendar): pure modules — expandRecurrence + moveSession + dragMachine + buildMonthGrid + storage

### PR DESCRIPTION
**Pull request by iwash-alt**
## Summary

Phase 1 (PR B) of the program calendar rework. All pure modules + storage helpers + the `buildMonthGrid` extension to the existing date-helpers. No UI, no `WorkoutContext` extensions — those live in PR D.

Builds on PR #395 (docs/spec/plan/mockups). Should merge after #395.

### What's in

- **`types.ts`** — `RecurrenceRule`, `SessionOverride`, `PlannedSession`, `MoveSource`. Composes with the existing `ScheduledWorkoutSession` and `ProgramSource`.
- **`recurrenceValidation.ts`** — shape guards used by `addRecurrenceRule` / `updateRecurrenceRule` (PR D). 8 tests.
- **`expandRecurrence.ts`** — the core. Expands one-off sessions + recurring rules with overrides into a date-sorted `PlannedSession[]` within a window. 26 tests covering one-off (5), weekly (4), biweekly anchor (3), skip (3), move with two-pass moved-in scan (3), endRule.count + endRule.date (3), edge cases (5: orphaned, sort stability, one-off + recurring same date, DST, year boundary).
- **`moveSession.ts`** — unified move reducer. One-off → in-place date update; recurring → move-override upsert (or collapse on move-back). 8 tests.
- **`dragMachine.ts`** — pure long-press → drag → drop state machine. 350ms lift threshold + 10px movement cancel. 10 tests.
- **`buildMonthGrid`** added to `src/features/workout/schedule/calendar.ts` — 6×7 Sun-anchored date grid. 5 tests appended to the existing `calendar.test.ts`.
- **Storage** — Zod schemas (`recurrenceRuleSchema`, `sessionOverrideSchema` as a discriminated union) + load/save helpers in `storage.ts` + barrel exports in `index.ts`. Falls back to `[]` on schema-parse failure (degrades quietly).

### Test plan

- [x] `npx tsc --noEmit` clean
- [x] `npx eslint` clean on touched files
- [x] `timeout 240 npx vitest run` — **150 files, 1948 tests, all green** (up from 146/1891 on main — +57 tests across the new modules)
- [x] No stray junk files staged (cleaned per CLAUDE.md hazard list)
- [ ] No visual change (no UI in this PR — orchestrator and components ship in PRs C and D)

### Architectural decisions baked in

- **String-comparable `YYYY-MM-DD` dates** throughout the hot path. UTC arithmetic helpers only in `expandRecurrence` and `buildMonthGrid`; the call surface is string-only. No DST artifacts.
- **Biweekly cadence anchors on `rule.startDate`** (not on the window). Moving the visible window doesn't change which weeks emit.
- **`endRule.count` caps from `rule.startDate`** — pre-window occurrences consume the cap. Then we filter to the window. Required so that scrolling backward in the calendar doesn't show occurrences that should have been "burned" by completions before the window.
- **Move overrides have a two-pass treatment**: pass 1 handles natural-occurrence-in-window cases; pass 2 catches moves whose original date is outside the window but whose `newDate` lands inside (with a `wouldHaveEmitted` guard against stale overrides).
- **No-op moves return the same state reference** so React reference-equality short-circuits re-render.

🤖 Generated with [claude-flow](https://github.com/ruvnet/claude-flow)

### CODE REVIEW

File: src/features/workout/program-calendar/calendar/moveSession.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      ),
+    };
+  }
+
+  // ── Recurring path ───────────────────────────────────────────────────────
+  const existingIdx = state.sessionOverrides.findIndex(
+    (o) =>
+      o.ruleId === source.ruleId
+      && o.originalDate === source.originalDate
+      && o.kind === 'move',
```
[the remark this answers is not in the document]
Comment (iwash-alt) [reply src/features/workout/program-calendar/calendar/moveSession.ts]: Fixed in c0a99f8. Dropped the `kind === 'move'` filter from the existing-override lookup so any prior override on `(ruleId, originalDate)` is treated as the single slot for that occurrence. Move-back-to-original now removes whatever override is there (move or skip); a new move with a prior skip replaces the skip in-place with the new move (no contradictory state); a new move with a prior move updates `newDate` in-place (id preserved). Three regression tests added in `moveSession.test.ts`: skip→move replaces (no contradiction), move-back clears a prior skip, and the full skip→move→move-back sequence ends with zero overrides — the original bug scenario from your review. All 4 pure-module suites green (55 tests, up from 52).
